### ISSUE TITLE
CopyPaste doesn’t take macros and in some cases should

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#3621]
CopyScene doesn’t copy macros (bipolar, name, routing) etc... and that makes sense. It shouldn’t do that because macros are per instance not per scene.

Except there’s a special case of : copy a scene, load an init patch, paste a scene

In that case the fact that copy and paste don’t copy and paste macros seems wierd. Looks t this screeenshot from JL.

![image](https://user-images.githubusercontent.com/13387561/104462332-73161780-557e-11eb-9c1d-5c77fe799e18.gif)

So in that case we need to do something special. 

It’s definitely the case that copy should snap all those values But when to paste them requires a bit of thought.

“Always” may be the right answer, by the way. If you haven’t changed stuff it is a no-op. But the idea that pasting from A into B could change something global is a bit odd (copy A, modify macro, paste B, macro changes. Is that right? I’m not sure).

So also you could do something where only this case is supported; Namely copy and load and paste pastes macros. (If your macro copy doesn’t match etc...)

But requires a bit of though.

**Comment by baconpaul** [comment]
The scene-to-scene copy of macros is also a problem; the modulation should at least apply into the B-side.

Basically “macro copy and paste doesn’t work at all and should do the intuitive and correct thing, considering the tricky non-scene-alignment of the controllers vs the scene-alignment of the modulators”

**Comment by mkruselj** [comment]
Sounds to me like cleanest way would be a direct option "Copy with macros", then paste pastes whatever is in the clipboard?

**Comment by baconpaul** [comment]
Yeah I think that’s right but there’s the A to b modulation state as well I need to consider
